universal-login-sdk/src/core/services/WalletStorageService.ts

```diff
@@ -1,36 +1,47 @@
+import {providers} from 'ethers';
 import {asAnyOf, asObject, asString, cast} from '@restless/sanitizers';
 import {ApplicationWallet, asExactly, SerializableFutureWallet} from '@unilogin/commons';
 import {WalletStorage, SerializedWalletState, SerializedDeployingWallet} from '../models/WalletService';
 import {IStorageService} from '../models/IStorageService';
 import {StorageEntry} from './StorageEntry';
 
-const STORAGE_KEY = 'wallet';
+const DEPRECATED_STORAGE_KEY = 'wallet';
 
 export class WalletStorageService implements WalletStorage {
   private storage: StorageEntry<SerializedWalletState>;
+  private readonly STORAGE_KEY: string;
 
-  constructor(private storageService: IStorageService) {
+  constructor(private storageService: IStorageService, chainName: string) {
+    this.STORAGE_KEY = `${DEPRECATED_STORAGE_KEY}-${chainName}`;
     this.storage = new StorageEntry(
-      STORAGE_KEY,
+      this.STORAGE_KEY,
       asSerializedState,
       storageService,
     );
   }
 
-  private migrateStoredState() {
+  async migrate() {
     try {
-      const data = this.storageService.get(STORAGE_KEY);
-      if (data === null) {
-        return;
+      const data = this.storageService.get(DEPRECATED_STORAGE_KEY);
+      if (data === null) return;
+      const {wallet} = cast(JSON.parse(data), asObject({
+        kind: asExactly('Deployed'),
+        wallet: asApplicationWallet,
+      }));
+      for (const key of ['ropsten', 'rinkeby', 'kovan', 'mainnet']) {
+        const network = key === 'mainnet' ? 'homestead' : key;
+        const provider = new providers.InfuraProvider(network);
+        const code = await provider.getCode(wallet.contractAddress);
```

Comment:
Maybe we should add new function to BlockchainService? 
```ts
isContract = async (address: string) =>
    '0x' !== await this.getCode(address)
```

then this code would look like: 
```ts
...
const blockchainService = new BlockchainService(new providers.InfuraProvider(network));
if (await blockchainService.isContract(wallet.contractAddress) {
...
```